theano/gof/opt.py

```diff
@@ -607,11 +607,20 @@ def process_node(self, fgraph, node):
         # properly.
         # The clients should at least contain `node` itself!
         if node.inputs:
-            assert len(node.inputs[0].clients) > 0
-            assert (node, 0) in node.inputs[0].clients
+            # Take the smallest clients list. Some ops like elemwise
+            # have optimization that put constant as the first inputs.
+            # As constant have in general more clients then other type of nodes
```

Comment:
Sorry for a late comment, a small correction, here it should be "than" and not "then".